### PR TITLE
feat: extends supported languages for code block

### PR DESCRIPTION
**Pull request by ldez**
- dockerfile
- properties (git-config, java-properties)
- makefile
- perl and perl6
- toml

### CODE REVIEW

File: grammars/asciidoc.cson [hunk 1]
```diff
@@ -778,6 +778,169 @@
...
+            'patterns': ['include': 'source.perl6']
+          }
+        ]
+      }
+
+      # Matches Toml AsciiDoc code blocks
+      #
+      # Examples
+      #
+      #   [source,rust]
```
Comment (nicorikken) [inline grammars/asciidoc.cson]: Typo?

Comment (ldez) [inline grammars/asciidoc.cson]: fixed!

**Comment by nicorikken** [comment]
All the source definitions seem to exist based on some GitHub code searches. Otherwise I find it hard to check. Maybe the `begin': '(^\\s*-{4,}\\s*$)'` regex can be put in a repository for reference, based on DRY. Either way the code-blocks involve a lot of unavoidable duplication.
Having the typo fix, this seems mergeable. Nice work! :+1:

**Comment by ldez** [comment]
I give you my trick for the languages support.

To get a list of all scope names registered in your Atom instance:

- open the Developer Tools Console: `Ctrl+Shift+I`
- execute the following:

```javascript
Object.keys(atom.grammars.grammarsByScopeName).sort().join('\n')
```

**Comment by ldez** [comment]
And the GitHub GFM languages code (base on [Ace](https://ace.c9.io/)): https://github.com/github/linguist/blob/master/lib/linguist/languages.yml

**Comment by nicorikken** [comment]
Golden!!!! Only one solution: `> CONTRIBUTING.adoc`

**Comment by mojavelinux** [comment]
> the code-blocks involve a lot of unavoidable duplication.

Is it possible use a match group in contentName and patterns fields? If so, we could consolidate these definitions down to a single grammar rule.

If that's not possible, perhaps we can create a script that generates this part of the file. The script would still need to be executed manually, but at least it would allow us to maintain a single rule and have all the permutations generated automatically.

**Comment by mojavelinux** [comment]
I think we should align with the Markdown grammars by using the content name `source.embedded.%s` where `%s` is the language name. wdyt? (perhaps a separate PR)

**Comment by mojavelinux** [comment]
> Golden!!!! Only one solution: > CONTRIBUTING.adoc

I like your thinking!

**Comment by ldez** [comment]
The current `contentName` with `source.%s` create a misleading with specific tools (like [linter-shellcheck](https://atom.io/packages/linter-shellcheck)).
Change to `source.embedded.%s` will fix this issue.

**Comment by mojavelinux** [comment]
Let's do it then!

**Comment by ldez** [comment]
It's already do! :wink:

**Comment by ldez** [comment]
once this PR merge, I'll do a big review on 'code block' grammar.

**Comment by mojavelinux** [comment]
Oops, I missed that. Super!

Let's proceed. :+1:

**Comment by nicorikken** [comment]
This new way indeed confirms to the [GitHub flavored markdown way](https://github.com/atom/language-gfm/blob/master/grammars/gfm.cson). Does this disable the language-specific linters in the code-blocks? This would certainly be helpful, because a typo in the JSON code-block has messed up my whole grammar parsing in the past. I'll get this merged next.

**Comment by ldez** [comment]
Linter are always working but some linters parse whole document, not only the code block, with the current naming.
I've prepare a PR with all languages blocks rewriting, I'm waiting  the merge of this PR.

**Comment by nicorikken** [comment]
I did a small check by checking for the embedded property in the cursor scope. @ldez thanks for the improvement

**Comment by mojavelinux** [comment]
@nicorikken & @ldez, welcome to the [contributors page](http://asciidoctor.org/contributors/) on asciidoctor.org! I'm so glad to have you as part of the team!

**Pull request merged**